File: go-backend/models/task_saved_search.go
package models

import "time"

// TaskSavedSearch is a user-saved task search (filter + sort + view mode)
// that is synced across devices.
type TaskSavedSearch struct {
	ID            int       `json:"id"`
	UserID        int       `json:"user_id"`
	Name          string    `json:"name"`
	FilterString  string    `json:"filter_string"`
	SortField     string    `json:"sort_field"`
	SortDirection string    `json:"sort_direction"`
	ViewMode      string    `json:"view_mode"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// CreateTaskSavedSearchParams is the request body for creating a saved search.
type CreateTaskSavedSearchParams struct {
	Name          string `json:"name"`
	FilterString  string `json:"filter_string"`
	SortField     string `json:"sort_field"`
	SortDirection string `json:"sort_direction"`
	ViewMode      string `json:"view_mode"`
}

// UpdateTaskSavedSearchParams is the request body for updating a saved search.
// All fields are optional; only provided fields are updated.
type UpdateTaskSavedSearchParams struct {
	Name          *string `json:"name"`
	FilterString  *string `json:"filter_string"`
	SortField     *string `json:"sort_field"`
	SortDirection *string `json:"sort_direction"`
	ViewMode      *string `json:"view_mode"`
}

File: zettelkasten-front/src/models/TaskSavedSearch.ts
import type { SortField, SortDirection, ViewMode } from '../types/taskPage';

/**
 * A user-saved task search, synced across devices via the backend.
 * Captures the full state of the Tasks page filter bar so a saved search
 * can be re-applied exactly: the filter string plus sort and view mode.
 */
export interface TaskSavedSearch {
  id: number;
  user_id: number;
  name: string;
  filter_string: string;
  sort_field: SortField;
  sort_direction: SortDirection;
  view_mode: ViewMode;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskSavedSearchParams {
  name: string;
  filter_string: string;
  sort_field: SortField;
  sort_direction: SortDirection;
  view_mode: ViewMode;
}

export interface UpdateTaskSavedSearchParams {
  name?: string;
  filter_string?: string;
  sort_field?: SortField;
  sort_direction?: SortDirection;
  view_mode?: ViewMode;
}
